estate/models/estate_property.py

```diff
@@ -0,0 +1,151 @@
+    garden = fields.Boolean(string="Garden", help="Property Garden")
+    garden_area = fields.Integer(string="Garden Area", help="Property Garden Area")
+    garden_orientation = fields.Selection(
+        string="Orientation",
+        selection=[("North", "North"), ("South", "South"), ("East", "East"), ("West", "West")],
+        help="Orientation help to define the garden orientation")
+
+    active = fields.Boolean(string="Active", default=True, help="Property is available")
+
+    state = fields.Selection(
+        string="State",
+        selection=[("new", "New"), ("offer_received", "Offer Received"), ("offer_accepted", "Offer Accepted"),
+                   ("sold", "Sold"), ("canceled", "Canceled")],
+        default="new",
+        help="State of the property advertisement")
+
+    # links
+    property_type_id = fields.Many2one('estate.property.type', domain="[('id', '!=', False)]", string="Type")
+    tag_ids = fields.Many2many('estate.property.tag', string="Tag")
+
+    salesperson_id = fields.Many2one('res.users', string='Salesman', default=lambda self: self.env.user,
+                                     help="Salesperson")
+    buyer_id = fields.Many2one('res.partner', string="Buyer", copy=False, help="Buyer Person")
+
+    offer_ids = fields.One2many('estate.property.offer', 'property_id', string='Offers', help='List of offers')
+
+    # campos computados
+    total_area = fields.Integer(string="Total Area (sqm)", compute='_compute_total_area')
+
+    best_price = fields.Float(string="Best Offer", compute='_compute_best_offer')
+
+    # campo compañia si se tienen 2 o más
+    company_id = fields.Many2one('res.company', string="Company", required=True, default=lambda self: self.env.company,
+                                 help="Company related to this property")
+
+    # constraints
+
+    _sql_constraints = [
+        ('check_expected_price', 'CHECK(expected_price > 0)', 'A property expected price must be strictly positive'),
+        ('check_selling_price', 'CHECK(selling_price >= 0)', 'A property selling price must be positive'),
+    ]
+
+    @api.depends('living_area', 'garden_area')
+    def _compute_total_area(self):
+        """
+            Compute and update total area.
+        """
+        for record in self:
+            record.total_area = record.living_area + record.garden_area
+
+    @api.depends('offer_ids.price')
+    def _compute_best_offer(self):
+        """
+            Compute and update best offer
+        """
+        for record in self:
+            try:
+                record.best_price = max(record.offer_ids.mapped('price'))
+            except ValueError:
+                record.best_price = None
+
+    @api.onchange('garden')
+    def _onchange_garden(self):
+        if self.garden:
+            self.garden_area = 10
+            self.garden_orientation = 'North'
+        else:
+            self.garden_area = 0
+            self.garden_orientation = ''
```

Comment:
aqui da lo mismo ponerle `''` o `False`, pero por como lo solemos hacer en odoo, yo le pondria `False`
```suggestion
            self.garden_orientation = False
```